### DISCUSSION TITLE
Visual studio 2022 - dúvidas sobre aula "Calculadora"

### DISCUSSION THREAD

**Discussion by diegomfs** [balta-io/2801#47] [Perguntas e Respostas] [answered]
Olá professor uso do VS 2022 ele está com nova versão e não mostra mais os "using" e o "main".

Quando executo o código:
>>>
Console.Clear();

    Console.WriteLine("Primeiro valor:  ");
    float v1 = float.Parse(Console.ReadLine());

    Console.WriteLine("Digite o Segundo Valor: ");
    float v2 = float.Parse(Console.ReadLine());
    // float e parse para converter o valor.

    Console.WriteLine(""); // Pular uma linha somente

    float resultado = v1 + v2;
    //Console.WriteLine("O resultado da soma é " + resultado);
    Console.WriteLine($"O resultado da soma é {resultado}");
    //Console.WriteLine($"O resultado da soma é {v1 + v2}");
    //Console.WriteLine("O resultado da soma é " + (v1 + v2));

    //Console.WriteLine(v1);
<<<<<

No editor normal e executo em dotnet run, ele processa e pede os numeros e da o resultado, tudo sem problema.

Porém, o "main" não mostra mais, e quando criei os metodos "static void soma" e o de "subtração" o dotnet run não executa nada, só repete os números. Acho que é pq coloquei dentro do "Main", como que faço para tirar (pois não vejo mais ele) e executar igual o seu e dar certo?

O visual já abre assim entende
 >>>
// See https://aka.ms/new-console-template for more information
Console.WriteLine("Hello, World!");

<<<
Obrigado

**Comment by niziul** [comment, marked as answer]
Daria pra você enviar um print do seu projeto aberto no vscode?

**Reply by diegomfs** [reply]
Utilizo o Visual Studio 2022 (não o Vs code).

**Reply by diegomfs** [reply]
[
![print_calculador](https://user-images.githubusercontent.com/65240134/154179787-218a1890-2532-499f-ac8c-66e718021a0c.jpg)
](url)

**Reply by niziul** [reply]
Olha no seu caso é só adicionar o escopo do programa. É necessário fazer isso em toda aplicação recém-criada. 

```
using System;

// O "namespace" é recomendado colocar o mesmo nome da aplicação
namespace AppCalculatorBaltaio
{
    // no nome da "class" você colocar o mesmo nome do arquivo ".cs"
    class Program
    {
        static void Main(string[] args)
        {
            // coloque aqui o contéudo do seu programa
            Console.WriteLine("Hello world!");
        }
    }
}
```

**Reply by andrebaltieri** [reply]
<3

**Comment by Gabopontes** [comment]
ola! eu tive o mesmo problema e descobri o que é, é bem simples na verdade.
parece que o .net 6 ele simplifica os códigos, provavelmente tu ta rodando a app nesse framework. da uma checada dps.

oq eu fiz para resolver foi na hora de criar  o projeto adicionar " --Framework net5.0" após "dotnet new console -o Calculator". até agora ainda não descobri jeito mais fácil de mudar os frameworks. acabei instalando o .net 6 por engano.
inclusive se alguem souber como troca de uma outra maneira ou como desinstala me diz ai! kkk

abraços!

**Comment by Dallonso** [comment]
O meu VCode tbm esta no .net 6.0 e não aparece o main e o using, mas esta funcionando normalmente.
Posta seu código completo com as funções para que possamos dar uma analisada.
Vou deixar o meu aqui para você testar pode ser?

```
// See https://aka.ms/new-console-template for more information
Menu();

static void Menu()
{
    Console.Clear();

    Console.WriteLine("Calculator - v0.0.0");
    Console.WriteLine("1 - Soma");
    Console.WriteLine("2 - Subtração");
    Console.WriteLine("3 - Divisão");
    Console.WriteLine("4 - Multiplicação");
    Console.WriteLine("5 - Sair");

    Console.WriteLine("-----------");
    Console.WriteLine("selecione a Operação:");

    short res = short.Parse(Console.ReadLine());

    switch (res)
    {
        case 1: Soma(); break;
        case 2: Subtracacao(); break;
        case 3: Divisao(); break;
        case 4: Multiplicacao(); break;
        case 5: System.Environment.Exit(0); break;
        default: Menu(); break;
    }
}

static void Soma()
{
    Console.Clear();

    Console.WriteLine("Primeiro Valor: ");
    float value1 = float.Parse(Console.ReadLine().Replace(".",","));

    Console.WriteLine("Segundo Valor: ");
    float value2 = float.Parse(Console.ReadLine().Replace(".",","));

    float result = value1 + value2;

    Console.WriteLine($"O valor da soma é de {result}");
    Console.ReadKey();
    Menu();
}

static void Subtracacao()
{
    Console.WriteLine("Primeiro Valor:");
    float value1 = float.Parse(Console.ReadLine().Replace(".",","));

    Console.WriteLine("Segundo Valor: ");
    float value2 = float.Parse(Console.ReadLine().Replace(".",","));

    float result = value1 - value2;

    Console.WriteLine($"O valor da subtração é de {result}");
    Console.ReadKey();
    Menu();
}

static void Divisao()
{
    Console.Clear();

    Console.WriteLine("Primeiro valor:");
    float value1 = float.Parse(Console.ReadLine().Replace(".",","));

    Console.WriteLine("Segundo Valor:");
    float value2 = float.Parse(Console.ReadLine().Replace(".",","));

    float result = value1 / value2;
    float rest = value1 % value2;

    Console.WriteLine($"O valor da divisão é {result}");
    Console.WriteLine($"O resto é {rest}");
    Console.ReadKey();
    Menu();

}

static void Multiplicacao()
{
    Console.Clear();

    Console.WriteLine("Primeiro valor:");
    float value1 = float.Parse(Console.ReadLine().Replace(".",","));

    Console.WriteLine("Segudo valor:");
    float value2 = float.Parse(Console.ReadLine().Replace(".",","));

    float result = value1 * value2;

    Console.WriteLine($"O valor da multiplicação é {result}");
    Console.ReadKey();
    Menu();
}
```

Posta o seu completo para que possamos dar uma olhada.
Abraço

**Comment by BrewertonSantos** [comment]
Olá @diegomfs e todos colegas.

## Sobre a diferença entre seu projeto e o apresentado na aula

Olá! A versão do .NET utilizada no curso é a Core 3.1, podemos confirmar isto na descrição dele:

![image](https://user-images.githubusercontent.com/55927647/154677783-f2bd0f47-d9b2-4f39-ad9c-4cc6c5cafe33.png)

Quando você utiliza o comando sugerido ao iniciar o projeto da calculadora você estará criando um projeto com a versão mais atual do .NET que você tiver instalada. Portanto neste caso a versão do projeto criado foi o .NET 6. 

Caso esteja conhecendo o framework e comandos, sugiro que desenvolva os projetos na mesma versão. Em um curso mais a frente na carreira isto será explicado para você.

Para criar um projeto do tipo `console app com a versão mais atual instalada do .NET` você utiliza este commando:

```console
dotnet new console -o nomedoprojeto
```

Já para `criar um projeto com uma versão específica do .NET` você pode fazer da seguinte forma:

```console
dotnet new console -o nomedoprojeto -f netcoreapp3.1
```

`Nota 1:` Verifique que após o nome do projeto inserimos um parâmetro -f para especificar a versão do framework.

## Explicando as diferenças entre versões

No .NET 6 aconteceram pequenas mudanças na estrutura do código para simplificar as cosias. Isso aconteceu com o `program.cs` que perdeu a antiga estrutura:

```csharp
using System;

namespace nomedoprojeto
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }
    }
}
```

Para se tornar assim:

```csharp
Console.WriteLine("Hello, World!");
```

Bem mais simples, certo?! Outras mudanças são:

1 - Namespaces não precisam mais ser declarados com chaves:

```csharp
namespace nomedoprojeto
{
    public class nomedaclasse { }
}
```

Podendo ser declarado desta forma agora:

```csharp
namespace nomedoprojeto;

public class nomedaclasse { }
```

E o uso dos usings agora não precisa ser declarado dentro de cada arquivo que usará a biblioteca. Agora podendo serem chamados com `global using` apenas uma vez e poderá ser chamado em qualquer parte da aplicação.

`Importante:` Você aprenderá sobre estes conceitos no curso `Fundamentos do ASP.NET 6` então não se apresse. Acompanhe o curso e absorva o que está sendo ensinado.
